src/services/codefixes/inferFromUsage.ts

```diff
@@ -875,45 +885,183 @@ namespace ts.codefix {
 
                 types.push(checker.createAnonymousType(/*symbol*/ undefined!, members, callSignatures, constructSignatures, stringIndexInfo, /*numberIndexInfo*/ undefined)); // TODO: GH#18217
             }
-            return types;
+            return types; // TODO: Should cache this since I HOPE it doesn't change

```

Comment:
like normal checking caching should be fine here. I'll do that in a followup PR.